.github/workflows/unit-tests.yaml

```diff
@@ -1,10 +1,5 @@
 name: Pull Request
-on:
```

Comment:
Is the intention that this runs every time there's a push to any branch or only pushes to the default branch?

Reply:
It's intentional to run on every push to every branch, mainly to mirror what was done in our `contract-tests.yaml` workflow. I don't have a preference here, I'm okay with changing it back.

Reply:
Oh, yeah we should change it since it does cost $$$ And I don't think it's really necessary

Reply:
As in, I'll push things just so my laptop isn't the single point of failure for my work, I would never check the CI for that. I don't know if that's just me.